braid/_braid.c

```diff
@@ -3213,6 +3213,13 @@ _braid_FRefine(braid_Core   core,
    /* Initialize new hierarchy */
    _braid_CoreElt(core, gupper)  = f_gupper;
    _braid_CoreElt(core, nrefine) += 1;
+   braid_Int newrefine = _braid_CoreElt(core, nrefine);
+   // If we have hit max refinement, then limit the max levels
+   if(!(newrefine < max_refinements))
```

Comment:
I would rename newrefine to "numrefine"

Reply:
Those both sound reasonable, changed here: 5fdcc15c8c6672b09dd258bd0609d7e189efe63d

I mentioned this in my other comment: we probably need to make one other check. How do we detect when our refinement is halted because the step routine no longer calls SetRFactor() but we still haven't met either of the above conditions (max_refinements or tpoints_cutoff)?

Reply:
I'm not sure how to easily check if the user has stopped calling SetRFactor, either.  You could pick a place in the code, and do an AllReduce on every processors's rfactors.  If the sum is npoints, then the user has stopped calling rfactor (at least for this iteration, it's possible the user could call rfactor the next iteration).

But, Is this something that we really need to check, in that this feature is designed for a very specific use-case?  In other words, is this pull request (minus a little regression test) done?

What do you think Rob? @rfalgout